opensquirrel/exporter/quantify_scheduler_exporter.py

```diff
@@ -61,14 +60,23 @@ def visit_bloch_sphere_rotation(self, g: BlochSphereRotation) -> None:
             self.schedule.add(quantify_scheduler_gates.Rz(theta=theta, qubit=self._get_qubit_string(g.qubit)))
             return
 
-        raise _unsupported_gates_exception
+        raise UnsupportedGatesError(
+            f"Cannot export circuit: it contains unsupported gates. The following gate could not be exported: {g}. "
+            "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
+        )
 
     def visit_matrix_gate(self, g: MatrixGate) -> None:
-        raise _unsupported_gates_exception
+        raise UnsupportedGatesError(
+            f"Cannot export circuit: it contains unsupported gates. MatrixGates are not supported. "
+            "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
+        )
```

Comment:
Can we make a constructor for UnsupportedGatesError receiving a string?

The constructor would be something like:
```
    def __init__(self, gate_type: str):
        self.message =  f"Cannot export circuit: it contains unsupported gates. The following gate could not be exported: {gate_type}"
            "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
```

Then, from `visit_matrix_gate`, we would call `raise UnsupportedGatesError('MatrixGate')`.

Reply:
In this case the error message is specifically tailored towards Quantify Scheduler, but the error type `UnsupportedGatesError ` is more general. I like the idea of making a more standardized and easier way to use the `Exception`, but I think the place should be in the `_ScheduleCreator` (which we might want to rename to `_QuantifyScheduleCreator`).

So, something like
```python
class _ScheduleCreator:
  # ----------------
  # Other methods
  # ----------------

  @staticmethod
  def _raise_unsupported_gates_error(gate_type: Any) -> None:
    raise UnsupportedGatesError(
            f"Cannot export circuit: it contains unsupported gates. The following gate could not be exported: {gate_type}"
            "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
        )
```

Reply:
Hm... I think we should decouple Error classes from Scheduler classes.

I see OK to have a generic UnsupportedGateError, and maybe then a QSUnsupportedGateError.

The generic class could always raise `f"Cannot export circuit: it contains an unsupported gate: {message}"`, then QSUnsupportedGateError could raise `"{gate_type}. Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)`.

But the code would just do `raise QSUnsupportedGateError(gate_type)`.

Reply:
I made a new implementation were `UnsupportedGateError` is now easily initialized as you suggested. If it is raised by the exporter, than it will now raise an `ExportError` from the `UnsupportedGateError` with a message tailored towards the specific exporter (we currently have only 1)

Reply:
Thanks!